src/bun.js/test/expect.zig

```diff
@@ -1377,6 +1377,282 @@ pub const Expect = struct {
         return .zero;
     }
 
+    pub fn toContainEntry(
+        this: *Expect,
+        globalObject: *JSGlobalObject,
+        callFrame: *CallFrame,
+    ) JSValue {
+        defer this.postMatch(globalObject);
+        const thisValue = callFrame.this();
+        const arguments_ = callFrame.arguments(1);
+        const arguments = arguments_.slice();
+
+        if (arguments.len < 1) {
+            globalObject.throwInvalidArguments("toContainEntry() takes 1 argument", .{});
+            return .zero;
+        }
+
+        incrementExpectCallCounter();
+
+        const expected = arguments[0];
+        if (!expected.jsType().isArray()) {
+            globalObject.throwInvalidArgumentType("toContainEntry", "expected", "array");
+            return .zero;
+        }
+        expected.ensureStillAlive();
+        const value: JSValue = this.getValue(globalObject, thisValue, "toContainEntry", "<green>expected<r>") orelse return .zero;
+
+        const not = this.flags.not;
+        var pass = false;
+
+        if (!value.isUndefinedOrNull()) {
+            const key = expected.getIndex(globalObject, 0);
```

Comment:
```suggestion
            const key = expected.getIndex(globalObject, 0);
            if (globalObject.hasException()) return .zero;
```
Each `getIndex` call will need to check for exceptions. It's possible for the array to throw
```js
var obj = { a: 1 };
var arr = ["a"];
Object.defineProperty(arr, 1, {
    get: function() {
        throw new Error("error");
    }
    enumerable: true,
}
expect(obj).toContainEntry(arr);
```